cpp/include/cudf/utilities/traits.hpp

```diff
@@ -120,8 +127,7 @@ constexpr inline bool is_timestamp() {
  * @return false `type` is not a timestamp
  *---------------------------------------------------------------------------**/
 constexpr inline bool is_timestamp(data_type type) {
-  return cudf::exp::type_dispatcher(
-      type, [](auto dummy) { return is_timestamp<decltype(dummy)>(); }, 0);
+  return cudf::exp::type_dispatcher(type, is_timestamp_impl{});
```

Comment:
isn't this nicer?

```
return cudf::exp::type_dispatcher(type, [](auto object) {
  return is_timestamp<decltype(object)>() 
});
```

Reply:
Sadly it doesn't work because `string_view` breaks it. 

```
[](auto dummy) { return is_timestamp<decltype(dummy)>(); }, 0);
```

You need to pass in an argument for `value`, which is why `0` was passed in, but that's an invalid value for when `string_view` is used. There's no way to make an `auto` lambda without an argument, so we have to use an explicit function object instead.